src/mgmt/srv_pool.c

```diff
@@ -210,34 +210,9 @@ ds_mgmt_create_pool(uuid_t pool_uuid, const char *group, char *tgt_dev,
 		    int domains_nr, uint32_t *domains)
 {
 	uuid_t				*tgt_uuids = NULL;
-	d_rank_list_t			*filtered_targets = NULL;
-	d_rank_list_t			*pg_ranks = NULL;
-	uint32_t			pg_size;
 	int				rc;
 	int				rc_cleanup;
 
-	/* Sanity check targets versus cart's current primary group members.
-	 * If any targets not in PG, flag error before MGMT_TGT_ corpcs fail.
-	 */
-	rc = crt_group_size(NULL, &pg_size);
-	D_ASSERTF(rc == 0, ""DF_RC"\n", DP_RC(rc));
-	pg_ranks = d_rank_list_alloc(pg_size);
-	if (pg_ranks == NULL) {
-		rc = -DER_NOMEM;
-		D_GOTO(out, rc);
-	}
-	rc = d_rank_list_dup(&filtered_targets, targets);
-	if (rc) {
-		rc = -DER_NOMEM;
-		D_GOTO(out, rc);
-	}
-	/* Remove any targets not found in pg_ranks */
-	d_rank_list_filter(pg_ranks, filtered_targets, false /* exclude */);
-	if (!d_rank_list_identical(filtered_targets, targets)) {
-		D_ERROR("some ranks not found in cart primary group\n");
-		D_GOTO(out, rc = -DER_OOG);
-	}
```

Comment:
@kccain: I wound up removing this because I found that pool creates after a leadership change would get stuck in a neverending retry loop of -DER_OOG errors. I don't fully understand why that is. I seem to recall that you had added this check as more of a belt-and-suspenders approach to fixing some other issue, so maybe it's not really necessary?

Reply:
This was a discussion previously in https://jira.hpdd.intel.com/browse/DAOS-5531 and in the PR that added this code https://github.com/daos-stack/daos/pull/3492 .

The MGMT_TGT_CREATE and DESTROY RPCs had lots of problems when they were provided a set of destination ranks only some of which were in the cart primary group, i.e., when the corpcs failed with DER_OOG themselves. So the above patch was developed as a way to proactively avoid it before engaging the corpcs. Since then I haven't attempted to present those conditions to the corpcs so I can't say definitively if the code is still necessary - my guess is it is needed.

This was seen to have happened in 2 cases
1) the conditions resulting in filing of DAOS-5531 itself (the control plane MS in the drpc providing a list of ranks - all ranks in the system at the time of the pool create request), only a subset of which at that moment were in the cart PG; i.e., ranks known to the MS but not yet to the ioserver/cart.
2) another condition in which a pool create request comes to the MS with a user-specified rank list, some of which may be invalid (e.g., including a rank not known to both the MS and the ioserver/cart).

Reply:
Could the -DER_OOG errors be related to the assignment to svc.lastMapVer at line 354 being too early (before making sure the resp status is OK)? Or, does a new MS leader proactively trigger an initial group update? Just some perhaps uneducated thoughts...

Reply:
@kccain: Hmm. OK. I've reinstated the check. I've also added some logic to force an increment to the group map version before triggering a group update after a new leader steps up. I had been relying on the fact that leadership changes usually coincide with rank (re-)joins, but I think it's probably best to make this explicit. Testing on the io500 cluster this morning seemed to indicate that the change doesn't hurt anything, at least. We'll see how it does in CI.

Reply:
@liw: That variable is is used to prevent a duplicate group update which can happen in the rare event that an asynchronous group update request was made while a batch of join requests is being serviced. We do check that the request succeeded before making the assignment, fwiw. In any case, I have added logic to explicitly do a group update with an incremented map version after a new leader steps up, which I suspect is most correct anyhow.